### PR TITLE
TEST/FLUSH: enable flush test on self transport

### PR DESCRIPTION
**Pull request by shssf**
- uct_flush_test changed to use sender()/receiver() functions instead
  class variables
- timeouts added for potential hung loops (gtest mustn't hung)

Ignored for this PR
- found ud_mlx5/uct_flush_test.am_pending_flush_nb failed from time to
  time. This was a gtest hung without timeout. The issue is in progress
  loop at line test_flush.cc:354 (currently with timeout)

### CODE REVIEW

File: test/gtest/uct/test_flush.cc [hunk 1]
```diff
@@ -271,21 +284,22 @@ UCS_TEST_P(uct_flush_test, am_flush_ep_nb) {
...
+     /* Send until resources are exhausted or timeout in 1sec*/
      unsigned count = 0;
+     ucs_time_t loop_end_limit = ucs_get_time() + ucs_time_from_sec(1.0);
      ssize_t packed_len;
      for (;;) {
-         packed_len = uct_ep_am_bcopy(m_sender->ep(0), AM_ID, pack_cb,
+         packed_len = uct_ep_am_bcopy(sender().ep(0), AM_ID, pack_cb,
                                       (void*)&sendbuf);
-         if (packed_len == UCS_ERR_NO_RESOURCE) {
+         if (packed_len == UCS_ERR_NO_RESOURCE || ucs_get_time() > loop_end_limit) {
```
Comment (yosefe) [inline test/gtest/uct/test_flush.cc]: code style, please surround each expression with ()

**Comment by yosefe** [comment]
@shssf looks ok except minor comment. i also saw ud failure here: #879

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2135/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2136/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1486/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1487/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2140/ for details.

**Comment by yosefe** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2155/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1500/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shssf looks like problem after merging with master?
```
16:02:25 [ RUN      ] self/uct_flush_test.am_pending_flush_nb/0
16:02:26 /scrap/jenkins/scrap/workspace/hpc-ucx-pr/label/hpc-test-node/contrib/../test/gtest/uct/test_flush.cc:324: Failure
16:02:26 Error: Device is busy
16:02:26 
[  FAILED  ] self/uct_flush_test.am_pending_flush_nb/0, where GetParam() = self/self (1000 ms)
```

**Comment by shssf** [comment]
@yosefe I'm fixing this

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2159/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1504/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
:+1:

**Pull request merged**